### DISCUSSION TITLE
Formalize interaction between Fiber scheduler & Ractor cooperative scheduling

### DISCUSSION THREAD

**Discussion by Forthoney** [ioquatix/ioquatix#33] [Ideas]
I wanted to loop you in on a [discussion ](https://bugs.ruby-lang.org/issues/20276) on the prospect of defining the interaction between the Fiber scheduler and the blocking cooperative scheduling operations on Ractors - that is, Ractor.yield, Ractor.take, Ractor.receive. I anticipate there to be some trickiness involved because unlike threads, Ractors may actually be running in parallel, so we may need to synchronize access to the scheduler. I have a decent understanding of Ractor internals, but don't know much about the internals of the Fiber scheduler. 

From my limited knowledge, I believe we can use `Scheduler.block` and `unblock` to implement this, but I would really appreciate your input

**Comment by ioquatix** [comment]
I replied on the issue.
